deepchecks/nlp/metric_utils/token_classification.py

```diff
@@ -70,172 +65,15 @@ def get_scorer_dict(span_aligner: TSpanAligner,
-        """Get the precomputed transformed labels and predictions."""
-        return self.processed_labels, self.processed_predictions
-
-    def fit_transform(self, y_true: TTokenLabel, y_pred: 'TTokenPred'
-                      ) -> t.Tuple[t.List[t.List[str]], t.List[t.List[str]]]:
-        """Fit (if necessary) and transform the labels and predictions of a specific dataset."""
-        # If fit was called in the past on the same labels and predictions, can skip fitting
-        if (self._pred_hash != self._hash_token_annotations(y_pred)) or \
-                (self._label_hash != self._hash_token_annotations(y_true)):
-            self.fit(y_true, y_pred)
-        return self.get_transformed()
-
-    def _spans_to_token_list(self, y_true: TTokenLabel, y_pred: 'TTokenPred',
-                             ) -> t.Tuple[t.List[t.List[str]], t.List[t.List[str]]]:
-        """Convert from predictions and labels in span format to a list of token annotations.
-
-        Parameters:
-        -----------
-            y_true: TTokenLabel
-                Labels as provided by the user in the accepted deepchecks format (TTokenLabel, detailed in TextData)
-            y_pred: TTokenPred
-                Predictions as provided by the user in the accepted format (TTokenPred, detailed in _shared_docs.py)
-
-        Returns:
-        --------
-            A tuple of processed labels and processed predictions, each a list of lists of string annotations, as
-            detailed on the class docstring.
-        """
-        if not len(y_true) == len(y_pred):
-            raise DeepchecksValueError(f'Must have both a label entry and a prediction entry for each sample, but '
-                                       f'found {len(y_true)} label entries and {len(y_pred)} prediction entries')
-
-        labels = []
-        predictions = []
-        for sample_idx in range(len(y_true)):
-            # Sort label and predictions by the starting position of the span
-            sample_y_true = sorted(y_true[sample_idx], key=lambda x: x[1])
-            sample_y_pred = sorted(y_pred[sample_idx], key=lambda x: x[1])
-            labels.append([annotation[0] for annotation in sample_y_true])
-
-            # A list of the spans, sorted by their starting position
-            label_spans = [(annotation[1], annotation[2]) for annotation in sample_y_true]
-            pred_spans = [(annotation[1], annotation[2]) for annotation in sample_y_pred]
-
-            # Here we want to find the prediction spans that match specific label spans, and discard the rest.
-            # We want to do that in O(N*log(N)) - assuming both lists are of length N approximately. Thus, the use of
-            # np.searchsorted
-            if len(sample_y_true) > 0:
-                # Indices of where to place elements of label_spans in pred_spans so that pred_spans is still sorted.
-                sorted_label_in_pred_index = np.searchsorted(list(map(lambda x: x[0], pred_spans)),
```

Comment:
RIP

Reply:
Is this kept somewhere else? A gist or something?

Reply:
I have it locally, thought of uploading to drive. Other ideas?